mpc_planner_node:
    ros__parameters:
      type: mpc
      mpc:
        model_name: pointRobot
        n: 3
        time_horizon: 10
        time_step: 0.1
        slack: False
        interval: 1
        initialization: "current_state"
        constraints: 
          - JointLimitConstraints
          - InputLimitConstraints
        objectives:
         - 'GoalReaching'
        weights:
         w: 1.0
         wvel: [0.0, 0.0, 0.0, 1.0, 0.01]
         ws: 1e10
         wu: 0.02
         wobst: 0.5
        number_obstacles: 1
        control_mode: "vel"
      robot:
        collision_links:
         - base_link
        selfCollision:
         pairs:  []
        urdf_file: pointRobot.urdf
        root_link: ee_link
        end_link: base_link
        base_type: holonomic #diffdrive
      example:
        debug: False
        floating: True
